skops/hub_utils/_hf_hub.py

```diff
@@ -205,14 +238,69 @@ def recursively_default_dict() -> MutableMapping:
         config["sklearn"]["example_input"] = _get_example_input(data)
         config["sklearn"]["columns"] = _get_column_names(data)
     elif "text" in task:
-        if isinstance(data, list) and all(isinstance(x, str) for x in data):
-            config["sklearn"]["example_input"] = {"data": data[:3]}
-        else:
-            raise ValueError("The data needs to be a list of strings.")
+        error_message = "The data needs to be an iterable of strings."
+        try:
+            data_head = _head(data, n=3)
+            if _is_sequence_of_strings(data_head):
+                config["sklearn"]["example_input"] = {"data": data_head}
+            else:
+                raise ValueError(error_message)
+        except TypeError as e:
+            raise ValueError(error_message) from e
 
     dump_json(Path(dst) / "config.json", config)
 
 
+def _head(data: Iterable, n: int):
```

Comment:
At some point, we will probably have to factor out functions like `_validate_folder`, `_head`, `_is_sequence_of_strings` etc. to a utils module. It doesn't need to be in this PR though.